android/src/main/java/cl/json/social/InstagramStoriesShare.java

```diff
@@ -49,6 +49,10 @@ protected String getPlayStoreLink() {
     }
 
     private void shareStory(ReadableMap options) {
+        if (!this.hasValidKey("appId", options)) {
+            throw new IllegalArgumentException("appId was not provided.");
+        }
+
```

Comment:
I like the idea of failing fast for developers, and I see you are error checking here (for missing appId), but on the iOS side, not checking

I wonder if you could check in the javascript code and fail there with an error message, but slightly more descriptive, for example "Instagram Story share requires an appId based on Meta policy" or similar?

That way (in javascript) it would fail on both platforms, with a helpful message

Reply:
@mikehardy thanks for the support